cpp/src/arrow/flight/sql/server.h

```diff
@@ -85,6 +85,32 @@ struct ARROW_FLIGHT_SQL_EXPORT PreparedStatementUpdate {
   std::string prepared_statement_handle;
 };
 
+/// \brief A bulk ingestion request
+struct ARROW_FLIGHT_SQL_EXPORT StatementIngest {
+  enum IngestMode {
```

Comment:
Move the enum to types.h and just have one definition for client/server?

Reply:
Thanks for this suggestion. The duplication was feeling awkward but I hadn't yet realized they shared that header file. I've made this change now.